### ISSUE TITLE
Error: : CommandLine Error: Option 'verify-dom-info' registered more than once!

### ISSUE THREAD

**Issue by KareemErgawy** [AnyDSL/impala#63]
This is not an issue with AnyDSL specifically but rather (I guess) a problem with linking against llvm libs (because of my system setup). I am asking for help in case somebody faced the same problem before.

Today, I had to install some new packages (bumblebee and cuda toolkit) to work on AnyDSL on Linux rather Windows like I used to. During the process, I guess some packages needed to install llvm libs. Running: `dnf list installed | grep llvm`, produces the following output:
```bash
llvm-libs.i686                         3.9.1-2.fc25             @updates        
llvm-libs.x86_64                       3.9.1-2.fc25             @updates
```

Now (using the standard AnyDSL installation), when I try to build something using impala (e.g. stincilla), I get the following error:
```make
[  2%] Generating gaussian.ll
: CommandLine Error: Option 'verify-dom-info' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
CMakeFiles/gaussian.dir/build.make:79: recipe for target 'gaussian.ll' failed
```

According to this (http://lists.llvm.org/pipermail/lldb-dev/2015-March/006852.html), this might be an issue of linking some llvm libs twice.

Does anyone have any pointers on how to resolve this issue?

**Comment by richardmembarth** [comment]
That's caused by transient dependencies due to linking LLVM twice into our binary. This is only a problem if you have static LLVM libraries (this is the default in our scripts). To fix this, you can
- use shared LLVM libraries
- revert the problematic commit cdl-saarland/rv@9096ff732e27b0ecc2b71dfe6bc74e42a77a2ec8
Reverting the commit will work on Linux (by coincidence), but fail to link libRV on macOS and Windows.

**Comment by richardmembarth** [comment]
Fixed in cdl-saarland/rv@41620d398a74325a22258078fb01b426afa0cd1a.
